portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/aui/ScriptData.java

```diff
@@ -95,6 +95,28 @@ public void writeTo(Writer writer) throws IOException {
 		_writeAuiModulesTo(writer);
 	}
 
+	public void writeTo(Writer writer, String portletId) throws IOException {
+		if (Validator.isNull(portletId)) {
+			writeTo(writer);
+
+			return;
+		}
+
+		PortletData portletData = _portletDataMap.get(portletId);
+
+		if (portletData == null) {
+			return;
+		}
+
+		writer.write("<script type=\"text/javascript\">\n// <![CDATA[\n");
+
+		portletData._rawSB.writeTo(writer);
+
+		_writeEs6ModulesTo(writer);
```

Comment:
@pavel-savinov although I don't know what the logic do here, it just looks fishy. You only write out the selected PortletData._rawSB out, but writing out all portlets' Es6Modules and AuiModules.
Is that on purpose?

Reply:
@shuyangzhou, we don't care about the requires included(Es6 and Aui modules), but we want to avoid including other portlets scripts and this check helps here.